qiskit_dynamics/models/generator_models.py

```diff
@@ -231,37 +261,64 @@ class GeneratorModel(BaseGeneratorModel):
     def __init__(
         self,
         operators: Array,
+        drift: Optional[Array] = None,
         signals: Optional[Union[SignalList, List[Signal]]] = None,
         frame: Optional[Union[Operator, Array, BaseFrame]] = None,
-        cutoff_freq: Optional[float] = None,
+        evaluation_mode: str = "dense_operator_collection",
     ):
         """Initialize.
 
         Args:
-            operators: A rank-3 Array of operator components.
+            operators: A rank-3 Array of operator components. If
+                a frame object is provided, each operator is assumed
+                to be in the basis in which the frame operator is
+                diagonal.
+            drift: Optional, constant terms to add to G. Useful for
+                frame transformations. If a frame, but not a drift,
+                is provided, will be set to -F. If both are provided,
+                the drift will be set to drift - F.
             signals: Specifiable as either a SignalList, a list of
-                     Signal objects, or as the inputs to signal_mapping.
-                     GeneratorModel can be instantiated without specifying
-                     signals, but it can not perform any actions without them.
+                Signal objects, or as the inputs to signal_mapping.
+                GeneratorModel can be instantiated without specifying
+                signals, but it can not perform any actions without them.
             frame: Rotating frame operator. If specified with a 1d
-                            array, it is interpreted as the diagonal of a
-                            diagonal matrix.
-            cutoff_freq: Frequency cutoff when evaluating the model.
+                array, it is interpreted as the diagonal of a
+                diagonal matrix. If provided, it is assumed that all
+                operators are in frame basis.
         """
-        self.operators = to_array(operators)
 
-        self._cutoff_freq = cutoff_freq
+        # initialize internal operator representation
+        self._operator_collection = None
+        self._operators = Array(np.array(operators))
+        self._drift = None
+        self.drift = drift
+        self.evaluation_mode = evaluation_mode
+
+        # set frame and transform operators into frame basis.
+        self._frame = None
+        self.frame = Frame(frame)
 
         # initialize signal-related attributes
         self._signals = None
         self.signals = signals
 
-        # set frame
-        self.frame = frame
+    @property
+    def operators(self) -> Array:
+        return self._operators
 
-        # initialize internal operator representation in the frame basis
-        self.__ops_in_fb_w_cutoff = None
-        self.__ops_in_fb_w_conj_cutoff = None
+    @property
+    def hilbert_space_dimension(self) -> int:
+        return self._operators.shape[-1]
+
+    @property
+    def evaluation_mode(self) -> str:
+        return super().evaluation_mode
```

Comment:
Is this necessary if you're just straight up calling `super().evaluation_mode`? I think this should just be inherited.

Reply:
It really is not necessary–in fact, I'd like not to have it. It's only there because I haven't found a way to use the @evaluation_mode.setter decorator in this class to override the setter method without either 1) also overriding the getter method, with something like this, or 2) getting some linting "error" (not sure if it will actually error if I try to run it, or if I can just add a #pylint ignore or something)